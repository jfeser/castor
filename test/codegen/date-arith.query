select([(date("1997-07-01") + month(3)) as d, (date("1997-07-01") + day(90)) as dd], ascalar(0 as z))
