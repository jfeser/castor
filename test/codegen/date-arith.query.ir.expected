// Locals:
// hoisted0 : date (persists=false)
// hoisted1 : date (persists=false)
// hoisted2 : date (persists=false)
// hoisted3 : date (persists=false)
// hoisted4 : tuple[date, date] (persists=false)
fun printer () : void {
    hoisted0 = 1997-07-01;
    hoisted1 = hoisted0 + 3;
    hoisted2 = 1997-07-01;
    hoisted3 = hoisted2 + 90;
    hoisted4 = (hoisted1, hoisted3);
    print(tuple[date, date], hoisted4);
}
// Locals:
// hoisted5 : date (persists=false)
// hoisted6 : date (persists=false)
// hoisted7 : date (persists=false)
// hoisted8 : date (persists=false)
// hoisted9 : tuple[date, date] (persists=false)
fun consumer () : void {
    hoisted5 = 1997-07-01;
    hoisted6 = hoisted5 + 3;
    hoisted7 = 1997-07-01;
    hoisted8 = hoisted7 + 90;
    hoisted9 = (hoisted6, hoisted8);
    consume(tuple[date, date], hoisted9);
}
